source/windows/process.c

```diff
@@ -14,8 +14,48 @@
  */
 #include <aws/common/process.h>
 
+#include <aws/common/string.h>
 #include <process.h>
+#include <stdio.h>
+#define MAX_BUFFER_SIZE (2048)
 
 int aws_get_pid(void) {
     return _getpid();
 }
+
+struct aws_run_command_result *aws_run_command(struct aws_run_command_options options) {
+    FILE *output_stream;
+    char output_buffer[MAX_BUFFER_SIZE];
+    struct aws_run_command_result *result = aws_run_command_result_new(options.allocator);
+    if (!result) {
+        return NULL;
+    }
+
+    output_stream = _popen(aws_string_c_str(options.command), "r");
```

Comment:
[debatable] The posix & windows implementations of `aws_run_command()` are non-trivial and implemented twice. The only difference is the "_" in front of popen and pclose

I know it's always a judgement call with multi-platform code, but a single implementation with an ifdef'ed popen call seems much more practical

```
#ifdef _WIN32
output_stream = _popen(options.command, "r");
#else
output_stream = popen(options.command, "r");
#endif
```

Reply:
I think I can move them to process_common.c, which makes more sense.